app/src/features/FhirResourceTree/resourceTreeSlice.ts

```diff
@@ -23,32 +23,58 @@ type ResourceTreeSliceState = {
 
 const initialState: ResourceTreeSliceState = {};
 
-export const getNodeById = (
-  id: string,
+export const getNode = (
+  get: "path" | "id",
+  path: string,
   root: ElementNode
 ): ElementNode | undefined => {
-  if (root.id === id) return root;
+  if (root[get] === path) return root;
   for (const next of root.children) {
-    const result = getNodeById(id, next);
+    const result = getNode(get, path, next);
     if (result) return result;
   }
   return undefined;
 };
 
+const addDataAttributes = (
+  dataAttributes: ElementNode[],
+  data: ElementNode
+) => {
+  dataAttributes.forEach((attribute) => {
+    if (attribute.definition.path) {
+      const node = getNode("path", attribute.definition.path, data);
+      if (node) {
+        node.children.push(attribute);
+        node.children.sort((a, b) => (a.path > b.path ? 1 : -1));
+      }
+    }
+  });
+};
+
```

Comment:
By directly including ElementNodes generated from Attributes in the Tree, all this shouldn't be needed anymore